### PR TITLE
Refactor join and add leave api

### PR DESCRIPTION
**Pull request by opxz7148**
- Refactor join by separate to other class
- Leave activity API

API URL change
Join - POST activities/join/<activity_id>/
Leave - DELETE activities/join/<activity_id>/

### CODE REVIEW

**Comment by OmegaOoh** [comment]
> * Refactor join by separate to other class
> * Leave activity API
> 
> API URL change Join - POST activities/join/<activity_id>/ Leave - DELETE activities/join/<activity_id>/

Does this wrong?

**Comment by opxz7148** [comment]
> Why Change the API? Please provide the reasons.

Previously to join an activities front end need to sent POST req to activities/<activity_id>/. To add a leave handler I need to implement DELETE req handler inside activity detail handler class, which I think it's violate SRP because ActivityDetail class should concern only edit and serve activity detail. So I decide to separate handler relate to join & leave to other class. Also with this structure URL will be more descriptive.

File: frontend/src/views/ActivityDetail.vue
Comment (OmegaOoh) [inline frontend/src/views/ActivityDetail.vue:1]: Duplicate code, the createPostRequestdoes handle all of that

**Review by OmegaOoh — CHANGES_REQUESTED** [review]

File: frontend/src/views/ActivityDetail.vue
Comment (opxz7148) [reply frontend/src/views/ActivityDetail.vue:1]: Remove on lastest commit,

**Review by OmegaOoh — APPROVED** [review]
Codes look good.

**Pull request merged**